packages/investor-foxy/src/api/api.ts

```diff
@@ -0,0 +1,658 @@
+    )
+    if (!liquidityReserveContract) throw new Error('Not a valid contract address')
+
+    const estimatedGas = await liquidityReserveContract.methods
+      .removeLiquidity(amountDesired.toString())
+      .estimateGas({
+        from: userAddress
+      })
+    return bnOrZero(estimatedGas)
+  }
+
+  async estimateWithdrawGas(input: WithdrawInput): Promise<BigNumber> {
+    const { amountDesired, userAddress, contractAddress, type } = input
+    const stakingContract = this.foxyStakingContracts.find(
+      (item) => toLower(item.options.address) === toLower(contractAddress)
+    )
+    if (!stakingContract) throw new Error('Not a valid contract address')
+
+    const isDelayed = type === WithdrawType.DELAYED && amountDesired
+    const estimatedGas = isDelayed
+      ? await stakingContract.methods.unstake(amountDesired.toString(), true).estimateGas({
+          from: userAddress
+        })
+      : await stakingContract.methods.instantUnstake(true).estimateGas({
+          from: userAddress
+        })
+    return bnOrZero(estimatedGas)
+  }
+
+  async estimateDepositGas(input: EstimateGasTxInput): Promise<BigNumber> {
+    const { amountDesired, userAddress, contractAddress } = input
+    const stakingContract = this.foxyStakingContracts.find(
+      (item) => toLower(item.options.address) === toLower(contractAddress)
+    )
+    if (!stakingContract) throw new Error('Not a valid contract address')
+    const estimatedGas = await stakingContract.methods
+      .stake(amountDesired.toString(), userAddress)
+      .estimateGas({
+        from: userAddress
+      })
+    return bnOrZero(estimatedGas)
+  }
+
+  async estimateApproveGas(input: EstimateGasApproveInput): Promise<BigNumber> {
+    const { userAddress, tokenContractAddress, contractAddress } = input
+    const depositTokenContract = new this.web3.eth.Contract(erc20Abi, tokenContractAddress)
+    const estimatedGas = await depositTokenContract.methods
+      .approve(contractAddress, MAX_ALLOWANCE)
+      .estimateGas({
+        from: userAddress
+      })
+    return bnOrZero(estimatedGas)
+  }
+
+  async approve(input: ApproveInput): Promise<string> {
+    const {
+      accountNumber = 0,
+      dryRun = false,
+      tokenContractAddress,
+      userAddress,
+      wallet,
+      contractAddress
+    } = input
+    if (!wallet) throw new Error('Missing inputs')
+    const estimatedGas: BigNumber = await this.estimateApproveGas(input)
+    const depositTokenContract = new this.web3.eth.Contract(erc20Abi, tokenContractAddress)
+    const data: string = depositTokenContract.methods
+      .approve(contractAddress, MAX_ALLOWANCE)
+      .encodeABI({
+        from: userAddress
+      })
+    const nonce: number = await this.web3.eth.getTransactionCount(userAddress)
+    const gasPrice: string = await this.web3.eth.getGasPrice()
+
+    const txToSign = buildTxToSign({
+      bip44Params: this.adapter.buildBIP44Params({ accountNumber }),
+      chainId: 1,
```

Comment:
if this thing takes a network constructor it needs to be used here, otherwise make it a private protected class variable